src/plot.js

```diff
@@ -437,7 +437,7 @@ function maybeMarkFacet(mark, topFacetState, options) {
   // top-level facet option.
   if (
     (groups.size > 1 || (groups.size === 1 && channels.fx && channels.fy && [...groups][0][1].size > 1)) &&
-    arrayify(mark.data)?.length === data.length
+    (arrayify(mark.data)?.length ?? NaN) === (data.length ?? NaN)
```

Comment:
How about instead testing that the data.length is positive before comparing?

```suggestion
    (data.length > 0 && arrayify(mark.data)?.length === data.length)
```